provider/pkg/autonaming/application.go

```diff
@@ -15,10 +15,26 @@ type AutoNamingConfig struct {
 	RandomSuffixMinLength int  `json:"randomSuffixMinLength"`
 }
 
+// EngineAutonamingConfiguration contains autonaming parameters passed to the provider from the engine.
+type EngineAutonamingConfiguration struct {
```

Comment:
Curious, why are we not reusing the autonaming structs and enum from pu-sdk?

Reply:
Great question. I started there but then went for keeping the `autonaming` module self-contained. It feels like a fairly complicated piece of "domain logic", so I wasn't sure I wanted to mix in RPC types. I'm not sure how much I bought by doing so - can be persuaded the other way.